### PR TITLE
reconcile internal changes to open source

### PR DESCRIPTION
**Pull request by dragonsinth**
I rebased all our internal solrman changes -> external repo.  Thoughts on proceeding here?  This would be a fast-forward of master.  Then we'd delete `gosolr` from the internal repo and replace with the go.mod line.

### CODE REVIEW

File: solrman/smmodel/core.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+type Core struct {
+	Name       string `json:"name"`
+	Collection string `json:"collection"`
+	Shard      string `json:"shard"`
+	Size       int64  `json:"size"` // in bytes
+	NumDocs    int64  `json:"num_docs"`
+
+	id           coreId
+	collectionId collectionId // the collection i belong to
```
Comment (EmilyBragg) [inline solrman/smmodel/core.go:28]: nit: i/I

**Comment by dragonsinth** [comment]
FYI: this is the diff between the new HEAD and what's currently checked into `mn`:
```patch
diff --git a/.travis.yml b/.travis.yml
index 2944f4e..1f72611 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -3,7 +3,7 @@ sudo: false
 
 matrix:
   include:
-    - go: 1.9
+    - go: 1.13
     - go: tip
 
 script:
diff --git a/go.mod b/go.mod
index 3d31112..179cdce 100644
--- a/go.mod
+++ b/go.mod
@@ -1,3 +1,5 @@
 module github.com/fullstorydev/gosolr
 
 go 1.13
+
+require github.com/samuel/go-zookeeper v0.0.0-20190923202752-2cc03de413da
diff --git a/go.sum b/go.sum
new file mode 100644
index 0000000..7bcd36c
--- /dev/null
+++ b/go.sum
@@ -0,0 +1,2 @@
+github.com/samuel/go-zookeeper v0.0.0-20190923202752-2cc03de413da h1:p3Vo3i64TCLY7gIfzeQaUJ+kppEO5WQG3cL8iE8tGHU=
+github.com/samuel/go-zookeeper v0.0.0-20190923202752-2cc03de413da/go.mod h1:gi+0XIa01GRL2eRQVjQkKGqKF3SF9vZR/HnPullcV2E=
diff --git a/solrman/smmodel/model_test.go b/solrman/smmodel/model_test.go
index 7a8fb04..6861a76 100644
--- a/solrman/smmodel/model_test.go
+++ b/solrman/smmodel/model_test.go
@@ -19,8 +19,6 @@ import (
 	"strconv"
 	"strings"
 	"testing"
-
-	"fs/fsmath"
 )
 
 func TestEmptyModel(t *testing.T) {
@@ -148,7 +146,7 @@ func TestLargeModel_EvacuatingNodes(t *testing.T) {
 
 func assertEquals(t *testing.T, expected []string, actual []Move) {
 	t.Helper()
-	for i, c := 0, fsmath.MaxInt(len(expected), len(actual)); i < c; i++ {
+	for i, c := 0, maxInt(len(expected), len(actual)); i < c; i++ {
 		if i >= len(actual) {
 			t.Errorf("at index: %d\nexpected: %s\n  actual: <nil>", i, expected[i])
 		} else if i >= len(expected) {
@@ -159,6 +157,14 @@ func assertEquals(t *testing.T, expected []string, actual []Move) {
 	}
 }
 
+func maxInt(a int, b int) int {
+	if a > b {
+		return a
+	} else {
+		return b
+	}
+}
+
 func createTestModel(data string, evacuatingNodes ...string) *Model {
 	var currentNode *Node
 	m := &Model{}
```

File: solrman/smmodel/core.go (hunk 1 above)
Comment (dragonsinth) [reply solrman/smmodel/core.go:28]: I'm just pulling things in as-is in the repo right now.  We can definitely follow up here with any fixes.

**Comment by dragonsinth** [comment]
Anyone want to stamp this?

**Pull request merged**